sql/colflow: TestVectorizedFlowDeadlocksWhenSpilling failed

sql/colflow.TestVectorizedFlowDeadlocksWhenSpilling [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5878083?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5878083?buildTab=artifacts#/) on master @ [0f100f3a09246bc80ef2c3119e4ec1c24ef36801](https://github.com/cockroachdb/cockroach/commits/0f100f3a09246bc80ef2c3119e4ec1c24ef36801):


```
=== RUN   TestVectorizedFlowDeadlocksWhenSpilling
    test_log_scope.go:162: test logs captured to: /artifacts/tmp/_tmp/136feb420592dee097d4750144c4b4cf/logTestVectorizedFlowDeadlocksWhenSpilling1701904069
    test_log_scope.go:80: use -show-logs to present logs inline
    vectorized_flow_deadlock_test.go:77: 
        	Error Trace:	vectorized_flow_deadlock_test.go:77
        	Error:      	Should be false
        	Test:       	TestVectorizedFlowDeadlocksWhenSpilling
    panic.go:500: -- test log scope end --
--- FAIL: TestVectorizedFlowDeadlocksWhenSpilling (11.52s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestVectorizedFlowDeadlocksWhenSpilling.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-18048

--- a/pkg/sql/colflow/vectorized_flow_deadlock_test.go
+++ b/pkg/sql/colflow/vectorized_flow_deadlock_test.go
@@ -36,7 +36,7 @@ func TestVectorizedFlowDeadlocksWhenSpilling(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.UnderStressRace(t, "the test is too slow under stressrace")
+	skip.UnderStress(t, "the query might take longer than timeout under stress making the test flaky")
 
 	vecFDsLimit := 8
 	envutil.TestSetEnv(t, "COCKROACH_VEC_MAX_OPEN_FDS", strconv.Itoa(vecFDsLimit))

